### PR TITLE
MM-4998 Adding LoginIdAttribute to allow LDAP users to change their login ID without losing their account

### PR DESCRIPTION
**Pull request by crspeller**
Enterprise PR: https://github.com/mattermost/enterprise/pull/286

This also adds a command to allow for the migration of users between different LDAP IdAttributes. (changes the AuthData)

### CODE REVIEW

File: cmd/commands/ldap.go [hunk 1]
```diff
@@ -22,9 +22,19 @@ var LdapSyncCmd = &cobra.Command{
 	RunE:    ldapSyncCmdF,
 }
 
+var LdapIdMigrate = &cobra.Command{
+	Use:     "idmigrate",
+	Short:   "Migrate LDAP IDAttribute to new value",
```
Comment (jwilander) [inline cmd/commands/ldap.go]: Minor type `IDAttribute` -> `IdAttribute`

**Comment by towolf** [comment]
Please, this should not be E20 only.

**Comment by jasonblais** [comment]
@tomitm This feature is included in all Mattermost editions where LDAP is supported (E10 and E20 today)

**Comment by towolf** [comment]
Ah, thanks for the Info. The 'changelog in progress' indicated otherwise here: https://github.com/mattermost/docs/pull/2019/files#diff-6d8bc7fdf1108df5a6734816051a32e3R48

**Comment by jasonblais** [comment]
@towolf Good catch! We'll fix that /cc @amyblais

**Pull request merged**